### PR TITLE
RUST-603 Lint documentation in CI

### PR DESCRIPTION
**Pull request by isabelatkinson**

### CODE REVIEW

**Review by saghm — APPROVED** [review]

**Review by patrickfreed — COMMENTED** [review]
Can we enable the [`missing_docs`](https://doc.rust-lang.org/rustdoc/lints.html#missing_docs) and `missing_crate_level_docs` lints? They're disabled by default so we currently aren't checking for them.

File: src/error.rs [hunk 1]
```diff
@@ -205,6 +205,7 @@ impl std::ops::Deref for Error {
 }
 
 /// The types of errors that can occur.
+#[allow(missing_docs)]
```
Comment (isabelatkinson) [inline src/error.rs:208]: This suppresses warnings for missing comments on all of the `message` fields -- doesn't seems super necessary/helpful to users to have an individual comment for each of these fields given they're fairly self-explanatory.

File: src/selection_criteria.rs [hunk 2]
```diff
@@ -93,12 +93,15 @@ pub type Predicate = Arc<dyn Send + Sync + Fn(&ServerInfo) -> bool>;
 /// option and will be sent to the server as an integer number of seconds.
 ///
 /// See the [MongoDB docs](https://docs.mongodb.com/manual/core/read-preference) for more details.
+#[allow(missing_docs)]
```
Comment (isabelatkinson) [inline src/selection_criteria.rs:96]: Ditto above for all of the `options` fields.

**Review by patrickfreed — APPROVED** [review]
LGTM!

**Pull request merged**